src/inc/genXplatEtw.pl

```diff
@@ -0,0 +1,762 @@
+use strict;
+use warnings;
+use XML::DOM;
+use FindBin;
+use Scalar::Util qw(looks_like_number);
+
+my %palDataTypeMapping =(
+#constructed types
+    "win:null"         =>" ",
+    "win:Int64"         =>"const __int64",
+    "win:ULong"         =>"const ULONG",
+    "win:count"         =>"*",
+    "win:Struct"       =>"const void*",
+#actual spec
+    "win:GUID"          =>"LPCGUID",
+    "win:AnsiString"    =>"LPCSTR",
+    "win:UnicodeString" =>"PCWSTR",
+    "win:Double"        =>"const double",
+    "win:Int32"         =>"const signed int",
+    "win:Boolean"       =>"const BOOL",
+    "win:UInt64"        =>"const unsigned __int64",
+    "win:UInt32"        =>"const unsigned int",
+    "win:UInt16"        =>"const unsigned short",
+    "win:UInt8"         =>"const unsigned char",
+    "win:Pointer"       =>"const void *"
+);
+
+my %seqDataTypeMapping =(
+    "win:GUID"          =>"GUID",
+    "win:Pointer"       =>"const void *"
+);
+
+
+my %lttngDataTypeMapping =(
+#constructed types
+    "win:null"          =>" ",
+    "win:Int64"         =>"const __int64",
+    "win:ULong"         =>"const ULONG",
+    "win:count"         =>"*",
+    "win:Struct"        =>"const int*",
+#actual spec
+    "win:GUID"          =>"const int*",
+    "win:AnsiString"    =>"const char*",
+    "win:UnicodeString" =>"const char*",
+    "win:Double"        =>"const double",
+    "win:Int32"         =>"const signed int",
+    "win:Boolean"       =>"const BOOL",
+    "win:UInt64"        =>"const unsigned __int64",
+    "win:UInt32"        =>"const unsigned int",
+    "win:UInt16"        =>"const unsigned short",
+    "win:UInt8"         =>"const unsigned char",
+    "win:Pointer"       =>"size_t"
+);
+
+my %ctfDataTypeMapping =(
+#constructed types
+    "win:Int64"         =>"ctf_integer",
+    "win:ULong"         =>"ctf_integer",
+    "win:count"         =>"ctf_sequence",
+    "win:Struct"        =>"ctf_sequence",
+#actual spec
+    "win:GUID"          =>"ctf_sequence",
+    "win:AnsiString"    =>"ctf_string",
+    "win:UnicodeString" =>"ctf_string",
+    "win:Double"        =>"ctf_float",
+    "win:Int32"         =>"ctf_integer",
+    "win:Boolean"       =>"ctf_integer",
+    "win:UInt64"        =>"ctf_integer",
```

Comment:
Actually the generated code has the specific datatype, i use the lookup-table instead of checking for each individual type in an elaborate if condition !!! see line 454 in updated script